addons/Media_Recorder/src_webpack/player/BasePlayer.jsm

```diff
@@ -17,8 +22,9 @@ export class BasePlayer extends Player {
 
     setRecording(source) {
         this.mediaNode.src = source;
-        this._getDuration()
-            .then(duration => this.onDurationChangeCallback(duration));
+        var promise = this._getDuration();
+        promise.then(duration => this.onDurationChangeCallback(duration));
```

Comment:
Ten kod można wrzucić w jeden block then:
promise.then( duration => {
  this.mediaDuration = duration;
  this.onDurationChangeCallback(duration));
});